samples/kafka/quickstart/producer/src/main/java/com/example/app/TestDataReporter.java

```diff
@@ -18,7 +20,12 @@ public void run() {
                 long time = System.currentTimeMillis();
                 System.out.println("Test Data #" + i + " from thread " + Thread.currentThread().getId());
                 final ProducerRecord<Long, String> record = new ProducerRecord<Long, String>(TOPIC, time, "Test Data #" + i);
-                producer.send(record);
+              producer.send(record, new Callback() {
+					public void onCompletion(RecordMetadata metadata, Exception exception) {
+					if (exception != null) {
+						System.out.println(exception);
+						System.exit(1);
```

Comment:
evaluate if you want to remove `system.exit()` - the api design pattern they are trying to expose - is to - handle errors & keep sending.

Reply:
yeah, I will remove the exit.